cs_util/cfis.py

```diff
@@ -0,0 +1,98 @@
+"""CFIS.
+
+:Name: cfis.py
+
+:Description: This script contains CFIS-specific methods.
+
+:Author: Martin Kilbinger <martin.kilbinger@cea.fr>
+
+"""
+
+
+import re
+import numpy as np
+from astropy import units
+from astropy import coordinates as coords
+
+
+class Cfis(object):
+    """Cfis
+
+    Class for CFIS image properties.
+
+    """
+    size = {'tile' : 0.5 * units.deg}
+
+
+def get_tile_number(tile_name):
+    """Get Tile Number.
+
+    Return tile number of given image tile name.
+
+    Parameters
+    ----------
+    str
+        tile name
+
+    Raises
+    ------
+    ValueError
+        if tile name does not match expected pipeline numbering scheme
+
+    Returns
+    -------
+    tuple
+        tile number for x and tile number for y
+
+    """
+    m = re.search(r'(\d{3})[\.-](\d{3})', tile_name)
+    if m is None or len(m.groups()) != 2:
+        raise ValueError(
+            f'Image name \'{tile_name}\' does not match tile name syntax'
+        )
+
+    nix = m.groups()[0]
+    niy = m.groups()[1]
+
+    return nix, niy
+
+
+def get_tile_coord_from_nixy(nix, niy):
+    """Get Tile Coord From Nixy.
+
+    Return coordinates corresponding to tile with number (nix,niy).
+
+    Parameters
+    ----------
+    nix : str or int
+        tile number for x, can be list
+    niy : str or int
+        tile number for y, can be list
+
+    See also
+    --------
+    get_tile_number_from_coord
+
+    Returns
+    -------
+    tuple
+        right ascension and declination
+
+    """
+    if not np.isscalar(nix):
+        # Transform to int
+        xi = np.array(nix).astype(int)
+        yi = np.array(niy).astype(int)
+    else:
+        xi = int(nix)
+        yi = int(niy)
```

Comment:
So, this isn't wrong but it's not as restrictive as it could be. Since you effectively have 3 possible input data types (`str`, `int` or `list`) you may want to limit the scope or perhaps raise a type error if the inputs don't match this. For example,  if someone inputs floats for `nix` and `niy` things will work (since they will get converted to integers) but you probably don't want it to as it may give some funny results.

Reply:
OK, I removed the `int` options, this is also more consistent with similar methods around tile IDs. It's still ok to allow for scalar and list input, right?

Reply:
Hey @martinkilbinger so, after re-reading, I believe my original point was mainly what do you want to happen if someone calls `get_tile_coord_from_nixy(1.5, 2.6)`, are you happy for `xi, yi = (1, 2)`, or would you rather the code raised an exception?

Reply:
Good point. Now I am only allowing `str` or list of `str` on input, and I added tests in the function, as well as unit tests to check whether an exception is raised for invalid input.